packages/backend-auth/src/userpool_access_policy_factory.ts

```diff
@@ -22,127 +14,99 @@ export class UserPoolAccessPolicyFactory {
   private policyCount = 1;
 
   /**
-   * Instantiate with the bucket to generate policies for
+   * Instantiate with the userpool to generate policies for
    */
   constructor(private readonly userpool: IUserPool) {
     this.stack = Stack.of(userpool);
   }
 
-  createPolicy = (actions: Readonly<Set<UserpoolAction>>) => {
-    if (actions.size === 0) {
+  createPolicy = (actions: AuthActions) => {
+    const policyActions: Set<string> = new Set();
+
+    actions.forEach((authAction) => {
+      const mappedAction = iamActionMap[authAction];
+
+      if (typeof mappedAction === 'string') {
+        policyActions.add(mappedAction);
+      } else {
+        mappedAction.forEach((action) => policyActions.add(action));
+      }
+    });
+
+    if (policyActions.size === 0) {
       throw new AmplifyFault('EmptyPolicyFault', {
         message: 'At least one action must be specified',
       });
     }
 
-    const statements: PolicyStatement[] = [];
-
-    actions.forEach((action) => {
-      statements.push(this.getStatement(action));
-    });
+    const policy = new Policy(
+      this.stack,
+      `${this.namePrefix}${this.policyCount++}`,
+      {
+        statements: [
+          new PolicyStatement({
+            actions: [...policyActions],
+            resources: [this.userpool.userPoolArn],
+          }),
+        ],
+      }
+    );
 
-    return new Policy(this.stack, `${this.namePrefix}${this.policyCount++}`, {
-      statements,
-    });
+    return policy;
   };
-
-  private getStatement = (action: UserpoolAction) =>
-    new PolicyStatement({
-      actions: actionPolicyMap[action],
-      resources: [this.userpool.userPoolArn],
-    });
 }
 
-const actionPolicyMap: Record<UserpoolAction, string[]> = {
-  create: [
-    'cognito-idp:ConfirmSignUp',
-    'cognito-idp:AdminCreateUser',
-    'cognito-idp:CreateUserImportJob',
-    'cognito-idp:AdminSetUserSettings',
-    'cognito-idp:AdminLinkProviderForUser',
-    'cognito-idp:CreateIdentityProvider',
+type IamActionMap = {
+  [action in ActionIam]: string;
+} & {
+  [amplifyActionSet in ActionMeta]: string[];
+};
+
+const iamActionMap: IamActionMap = {
+  manageUser: [
```

Comment:
This list still seems like too much for "manageUser". For example [AdminGetDevice](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminGetDevice.html) doesn't seem like it has anything to do with user management. Same goes for the other "device related" APIs. While those devices are associted with users, I would suggest a separate meta action for "userDeviceManagement" if / when there is demand